File: friendly_cars_frontend/src/service/InvoiceService.js
import axios from "axios";

const INVOICE_BASE_URL = "http://localhost:8080/api/v1/invoices";

class InvoiceService {
    createInvoice(invoice) {
        return axios.post(INVOICE_BASE_URL, invoice)
            .then(response => response.data)
    }

    getAllInvoices() {
        return axios.get(INVOICE_BASE_URL)
            .then(response => response.data)
    }

    getInvoiceById(invoiceId) {
        return axios.get(INVOICE_BASE_URL + `/${invoiceId}`)
            .then(response => response.data)
    }

    updateInvoice(invoiceId, newInvoice) {
        return axios.put(INVOICE_BASE_URL + `/${invoiceId}`, newInvoice)
            .then(response => response.data)
    }

    partialUpdateInvoice(invoiceId, newInvoice) {
        return axios.patch(INVOICE_BASE_URL + `/${invoiceId}`, newInvoice)
            .then(response => response.data)
    }

    deleteInvoice(invoiceId) {
        return axios.delete(INVOICE_BASE_URL + `/${invoiceId}`, invoiceId)
            .then(response => response.data)
    }
}

export default new InvoiceService()

File: cs_manager_system/src/main/java/com/csManager/csmanager/service/InvoiceService.java
package com.csManager.csmanager.service;

import java.util.List;
import java.util.Map;

import com.csManager.csmanager.entity.Invoice;
import com.csManager.csmanager.exception.ClientNotFoundException;
import com.csManager.csmanager.exception.InvoiceNotFoundException;

import org.springframework.http.ResponseEntity;

/**
 * InvoiceService
 */
public interface InvoiceService {
    ResponseEntity<List<Invoice>> getAllInvoices();
    ResponseEntity<Invoice> getInvoiceById(long invoiceId) throws InvoiceNotFoundException;
    ResponseEntity<List<Invoice>> getAllInvoicesByClientId(String clientId) throws ClientNotFoundException;
    ResponseEntity<Invoice> createInvoice(Invoice invoice, String clientId) throws ClientNotFoundException;
    ResponseEntity<Invoice> updateInvoice(long invoiceId, Invoice updatedInvoice) throws InvoiceNotFoundException;
    ResponseEntity<Invoice> partialUpdateInvoice(long invoiceId, Map<Object, Object> fields) throws InvoiceNotFoundException;
    ResponseEntity<?> deletInvoice(long invoiceId) throws InvoiceNotFoundException;;
}
